### DISCUSSION TITLE
Why does adding a new text component generate hundreds of ComponentTextNode2

### DISCUSSION THREAD

**Discussion by teknosains** [GrapesJS/grapesjs#6314] [Q&A]
I was trying to check on every component added using this event.

```javascript
// this code is inside React useEffect

editor.current.on('component:add', (component) => {
   console.log(component);
   component.setStyle({
     width: '100%',
     height: '100%'
   });
});
```

here's what printed into the console. hundreds of ComponentTextNode2 object. is this normal behaviour?

![image](https://github.com/user-attachments/assets/1a37103d-96c1-4e3c-8c45-0beb0956c5dd)

**Comment by danstarns** [comment]
@teknosains I added a test here that adds a text input and asserts a single `component:add` event. We have a pass on our end, could you provide a reproduction? It could be something with your react setup that's causing a component to be added. 

* https://github.com/GrapesJS/grapesjs/pull/6316

I see you have: 

```
// this code is inside React useEffect
```

So check that the effect is not causing a re-render.

**Reply by teknosains** [reply]
Thanks for the response. The useEffect is causing a normal re-render. And that behavior only occurs on Text component. It behaves normal on other components like image, columns, video etc.

Yap, let me prepare the reproduction asap
